packages/manager/apps/key-management-service/src/components/Credential/Create/CreateGeneralInformations.component.tsx

```diff
@@ -0,0 +1,106 @@
+import CreateGeneralInformationsValidity from './generalInformations/CreateGeneralInformationsValidity';
+import CreateGeneralInformationsCreationMethod from './generalInformations/CreateGeneralInformationsCreationMethod.component';
+
+type CreateGeneralInformationsProps = {
+  name: string;
+  setName: Dispatch<SetStateAction<string>>;
+  validity: number;
+  setValidity: Dispatch<SetStateAction<number>>;
+  description: string | null;
+  setDescription: Dispatch<SetStateAction<string | null>>;
+  csr: string | null;
+  setCsr: Dispatch<SetStateAction<string | null>>;
+};
+
+const CreateGeneralInformations = ({
+  name,
+  setName,
+  validity,
+  setValidity,
+  description,
+  setDescription,
+  csr,
+  setCsr,
+}: CreateGeneralInformationsProps) => {
+  const { t } = useTranslation('key-management-service/credential');
+  const { okmsId } = useParams();
+  const navigate = useNavigate();
+  const credentialNameError = validateCredentialName(name);
+  const credentialDescriptionError = validateCredentialDescription(description);
+  const credentialValidityError = validateValidityDate(validity);
+
+  return (
+    <div className="flex flex-col gap-7 md:gap-9">
+      <div className="flex flex-col gap-6 md:gap-8">
+        <Subtitle>
+          {t(
+            'key_management_service_credential_create_general_information_title',
+          )}
+        </Subtitle>
+        <CreateGeneralInformationsName
+          name={name}
+          setName={setName}
+          credentialNameError={credentialNameError}
+        />
+        <CreateGeneralInformationsDescription
+          description={description}
+          setDescription={setDescription}
+          credentialDescriptionError={credentialDescriptionError}
+        />
+        <CreateGeneralInformationsValidity
+          validity={validity}
+          setValidity={setValidity}
+          credentialValidityError={credentialValidityError}
+        />
+        <Subtitle>
+          {t(
+            'key_management_service_credential_create_general_creation_method_title',
+          )}
+        </Subtitle>
+        <CreateGeneralInformationsCreationMethod csr={csr} setCsr={setCsr} />
+        <div className="flex gap-4">
+          <OsdsButton
+            size={ODS_BUTTON_SIZE.md}
+            inline
+            variant={ODS_BUTTON_VARIANT.stroked}
+            color={ODS_THEME_COLOR_INTENT.primary}
+            onClick={() => {
+              navigate(`/${okmsId}${ROUTES_URLS.credentials}`);
+            }}
+          >
+            {t('key_management_service_credential_create_cta_cancel')}
+          </OsdsButton>
+          <OsdsButton
+            size={ODS_BUTTON_SIZE.md}
+            inline
+            color={ODS_THEME_COLOR_INTENT.primary}
+            disabled={
+              !!credentialNameError ||
+              !!credentialDescriptionError ||
+              !!credentialValidityError ||
+              undefined
```

Comment:
the submit button should be disabled when the user choose the "upload you own private key" but the textarea is still empty